From a656e8d667057de96900b33207798edbd211ec5c Mon Sep 17 00:00:00 2001
From: John Youn <johnyoun@synopsys.com>
Date: Thu, 19 May 2016 17:26:10 -0700
Subject: [PATCH 06/13] usb: dwc3: gadget: Don't prepare TRBs if no space

If trbs_left == 0, we don't have any space left in the TRB ring so don't
prepare anything.

Signed-off-by: John Youn <johnyoun@synopsys.com>
Signed-off-by: Felipe Balbi <felipe.balbi@linux.intel.com>
(cherry picked from commit 27fc6e2bd0f5b1d18e1eb0a42fd84797f21539cd)
Signed-off-by: Janusz Dziedzic <januszx.dziedzic@intel.com>

Change-Id: I8b95e3e1e072b1a40a7bbee7ff5add3d7f6f581a
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-34491
Reviewed-on: https://android.intel.com:443/519130
---
 drivers/usb/dwc3/gadget.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/usb/dwc3/gadget.c b/drivers/usb/dwc3/gadget.c
index 316d877..8f80639 100644
--- a/drivers/usb/dwc3/gadget.c
+++ b/drivers/usb/dwc3/gadget.c
@@ -918,6 +918,8 @@ static void dwc3_prepare_trbs(struct dwc3_ep *dep)
 	BUILD_BUG_ON_NOT_POWER_OF_2(DWC3_TRB_NUM);
 
 	trbs_left = dwc3_calc_trbs_left(dep);
+	if (!trbs_left)
+		return;
 
 	list_for_each_entry_safe(req, n, &dep->pending_list, list) {
 		unsigned	length;
-- 
1.9.1

